projects/components/src/navigation/navigation-list.component.ts

```diff
@@ -87,6 +91,26 @@ export class NavigationListComponent {
     return this.collapsed ? IconType.TriangleRight : IconType.TriangleLeft;
   }
 
+  public shouldShowHeader(index: number): Observable<boolean> {
+    const features = [];
+    for (let i = index + 1; i < this.navItems.length; i++) {
+      if (this.navItems[i].type !== NavItemType.Link) {
+        break;
+      }
+
+      const navLinkItemFeatures = (this.navItems[i] as NavItemLinkConfig).features;
+      features.push(isEmpty(navLinkItemFeatures) ? 'None' : navLinkItemFeatures);
+    }
+
+    return isEmpty(features.flat())
+      ? of(false)
+      : combineLatest(
+          features
```

Comment:
You shouldn't need to check feature flag state every time. One resolution per browser refresh is enough. Then we can just share the same result throughout (till another browser refresh)